### ISSUE TITLE
Crash at start : 1.86.2.24057 or 1.88

### ISSUE THREAD

**Issue by zzzazzz** [VSCodium/vscodium#1814]
Hi !
I did a `snap refresh` yesterday night. But today codium crashs at start.
`
installed:          1.86.2.24057            (375) 312MB classic`

- On Ubuntu 22.04.4 LTS (all updates ok)
- Wayland
- Gnome 42.9

The last line of the verbose output says  : 
`[100216:0226/105740.201982:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.`

[codium_output.log](https://github.com/VSCodium/vscodium/files/14403017/codium_output.log)

Thkx

**Comment by daiyam** [comment]
It seems to be an issue with the GPU driver (can't find the driver):
```
libGL error: MESA-LOADER: failed to open iris: /usr/lib/dri/iris_dri.so: Ne peut ouvrir le fichier d'objet partagé: Aucun fichier ou dossier de ce nom (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)
```

Since I've only tested it in VirtualBox, I can't test your issue...

But can you try:
- launch with `codium --disable-gpu`
- or remove `~/.config/VSCodium/GPUCache`

**Comment by zzzazzz** [comment]
Ok, i did the 2 tries :
`codium --disable-gpu --verbose`
or
removed `~/.config/VSCodium/GPUCache`
(removed too my whole `.config/VSCodium` folder too)
-> Badly same results.

I tried too change the driver from `x.org` to `nvidia-driver-535` proprietary, unfortunatly

Finally i did a `snap revert codium` to be able working today ;) If i can help to resolve this, i will.

**Comment by daiyam** [comment]
I think the solution can be found there: https://forum.snapcraft.io/t/opengl-broken-in-classic-core20-snap-on-22-04/29401

I will add it to the Insiders version and I will tell you where the snap will be so you can test it 😉

**Comment by zzzazzz** [comment]
Ok, great, thkx

**Comment by daiyam** [comment]
FYI, I've opened a topic on snapcraft: https://forum.snapcraft.io/t/mesa-issue-in-classic-core20-on-22-04/39129

**Comment by zzzazzz** [comment]
OK, if you need more logs outputs. I'll do.

**Comment by zzzazzz** [comment]
Hi,
Nothing new ? I'm seeing new versions being released but still force to use the 1.84.
The last line of the 1.88 output says :
```
[main 2024-04-06T12:00:42.878Z] resolveShellEnv(): skipped (VSCODE_CLI is set)
[26236:0406/140042.983060:ERROR:gpu_process_host.cc(993)] GPU process exited unexpectedly: exit_code=133
[26236:0406/140042.983074:WARNING:gpu_process_host.cc(1363)] The GPU process has crashed 4 time(s)
[26236:0406/140043.099404:ERROR:gpu_process_host.cc(993)] GPU process exited unexpectedly: exit_code=133
[26236:0406/140043.099421:WARNING:gpu_process_host.cc(1363)] The GPU process has crashed 5 time(s)
[26236:0406/140043.231682:ERROR:gpu_process_host.cc(993)] GPU process exited unexpectedly: exit_code=133
[26236:0406/140043.231698:WARNING:gpu_process_host.cc(1363)] The GPU process has crashed 6 time(s)
[26236:0406/140043.231708:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.

```
At each new release, i've to `sudo snap revert` (to 1.84)

**Comment by zzzazzz** [comment]
Hi,
I can't anymore use codium because i can't revert it, all other chanel disapeared. Help !

**Comment by daiyam** [comment]
Have you tried the `.deb`?

**Comment by daiyam** [comment]
or `sudo snap revert codium --revision 371` to revert to the `1.85.2.24019`?

**Comment by zzzazzz** [comment]
`erreur : cannot revert "codium": cannot find revision 371 for snap "codium"`
I try the .deb of the 1.85

**Comment by zzzazzz** [comment]
ok, the .deb of the 1.85 works ! Ouf !!!

**Comment by zzzazzz** [comment]
here the lines of the 1.85 where it crashed with the 1.86+ :

`[14412:0509/231546.222835:WARNING:gpu_process_host.cc(1319)] The GPU process has crashed 5 time(s)
[14412:0509/231546.494027:ERROR:gpu_process_host.cc(954)] GPU process exited unexpectedly: exit_code=133
[14412:0509/231546.494041:WARNING:gpu_process_host.cc(1319)] The GPU process has crashed 6 time(s)
[main 2024-05-09T21:15:46.510Z] resolveShellEnv(): skipped (VSCODE_CLI is set)
[14558:0509/231546.511574:WARNING:vaapi_wrapper.cc(806)] VAAPI video acceleration not available for disabled
[14412:0509/231546.555863:WARNING:gpu_process_host.cc(977)] Reinitialized the GPU process after a crash. The reported initialization time was 0 ms
`

**Comment by daiyam** [comment]
Does the crash also happened with the `.deb`?

**Comment by zzzazzz** [comment]
Nope, it works with `.deb`, the last message was just to compare logs between 1.86+ (crash) and 1.85 (works)

**Comment by zzzazzz** [comment]
Hi, I solved the problem installing last codium (1.89.1) with the flatpak way (i don't how and why ;)

**Comment by zzzazzz** [comment]
i did a clean install Ubuntu 24.04 from 0 due to a crash while upgrading it. Now all is working well…
Ubuntu 24.04, Wayland, VSCodium 1.95.3 snap
